charts/actions-runner-controller/values.yaml

```diff
@@ -33,12 +33,33 @@ authSecret:
   ### GitHub PAT Configuration
   #github_token: ""
 
+# Optionally create additional secrets required for your deployments
+extraSecrets: []
```

Comment:
Thanks! Sorry if I wasn't clear enough but I'd like the chart to contain features that are directly related to actions-runner-controller only.

If this key named this way, I believe it can be out of the scope of actions-runner-controller chart, as no values/templates in it is intended to allow arbitrary resources that doesn't directly related to actions-runner-controller itself.

For example, resources like ingress, service, etc created by this chart are tightly-coupled to actions-runner-controller. You can't create arbitrary ingress or service resources that doesn't connect to the actions-runner-controller deployment with this chart. Similary you can't create artibrary service account that isn't used by actions-runner-controller. But this `extraSecrets` allows creating secrets that aren't used by actions-runner-controller at all.

I guess- the only way to make it into this chart might be to name this so that it works like the `serviceAccount` key.

I mean, name it:

```suggestion
imagePullSecrets: []
```

and let the chart map all the names to `--runner-image-secret-name` like

```
        {{- range .Values.imagePullSecrets }}
        - "--runner-image-pull-secret-name={{ .name }}"
        {{- end }}
```

so that it might become clear that this key is only for defining image pull secrets for actions-runner-controller and nothing else.

// To do this, you also need to tweak the golang-side of `--runner-image-pull-secret-name` so that it can accept multiple secret names by repeating the flag

Thanks anyway for your efforts! I hope my comment makes sense.

Reply:
I've pushed one change with the feeedback 👍  [`fd881f7` (#921)](https://github.com/actions-runner-controller/actions-runner-controller/pull/921/commits/fd881f79771aa857dc198e391514e04e18db1184)

- I've removed for now the possibility to define the "content" of the secret, and I'll think it this could be added on a separate PR or maybe on a different chart.
- I've added support to specify +1 secret name via go cmdline
- I didn't use "imagePullSecrets" because that's already used to define the image pull secrets for the deployment and it could be confusing or we may want to have different image pull secrets for deployment vs runner. I'm using a string array `image.actionsRunnerImagePullSecrets` which can define all the default secret names that are going to be used in the go cmdline

Let me know what your thoughts are 👍

Reply:
I see that one integration test failed - do you know if that test is flaky by any chance? I ran them locally but all seemed to passed for me using this PR

![image](https://user-images.githubusercontent.com/55707366/146052126-ee4cf28b-a3b7-4faa-bbc9-4ba84ce7c559.png)

Reply:
This does seem like a test flake. Let me rerun it and see if it passes!